contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,249 @@
+If the annotation specifies so (users choice), the federation controller will create the additional service of type &#39;LoadBalancer&#39; against each pod replica.
+The service name for service, one each against the pod will be the same name as the pod identity.
+The LB service creation at federation has the same drawback as discussed in _design alternative 1_.
+The proposed solution is elaborated in section **Handling additional LB Services**.
+
+## Storage volumes
+
+There are two ways a federated statefulset can be assigned a persistent storage.
+
+1 - A node local volume attached to the container
+
+2 - A persistent volume claim from the cloud provider.
+
+Either of the 2 options above can be used/specified by the user to attach a persistent store to the pods.
+
+It is known, that as of now, the same persistent volume, even if can be used in a different cluster, k8s directly does not provide an API, yet, which can aid the same. In the absence of no direct way of quick migration of the storage data from one zone to another, or from one cloud provider to another if the case be, the proposal is to disallow migration of pods across clusters.
+
+## Scale up/Scale down
+
+When a federated statefulset is scaled up or down, the distribution of replicas will be recalculated and the replica numbers in each cluster will be updated independently.
+
+The behaviour of the scaling on being targeted using a **federated hpa** will be discussed in a separate design.
+
+## Handling additional LB services
+
+As listed in the approaches, the current option of exposing the statefulset pods across clusters is to assign a LB service to each replica pod.
+If the federated service controller is used (by creating the service in federation) to create these services, then as per the current behaviour the service will be created in each cluster. 
+This leads to exactly one unused service per statefulset per cluster.
+There are two alternatives to circumvent this problem:
+
+1 - Cluster affinity is introduced at least for services such that the controller creates the service only in needed cluster. (The need of cluster affinity and anti-affinity in general is discussed [elsewhere](https://github.com/kubernetes/kubernetes/issues/41442) also)
+
+2 - Federated statefulset controller handles the creation of cluster local services, rather than the federated service controller. It also will need to handle the dns records for each LB service into the federation (or cloud provider specific) public dns server.
+
+We propose using alternative 1 listed here, as it fits broader scheme of things and is more consistent with the user expectation of being able to query all needed resources from the federation control plane and is less confusing to use at the same time.
+
+# Conclusion
```

Comment:
With some applications, you can only create one pod at a time.  How does this design proposal maintain the ordinal order as defined by the stateful set contract?

Does this proposal define the capability of migrating from a regular stateful set to a federated stateful set?